sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionTokenHelper.java

```diff
@@ -154,7 +154,8 @@ static boolean tryParse(String sessionToken, ValueHolder<ISessionToken> parsedSe
     public static void validateAndRemoveSessionToken(RxDocumentServiceRequest request) {
         String sessionToken = request.getHeaders().get(HttpConstants.HttpHeaders.SESSION_TOKEN);
         if (!Strings.isNullOrEmpty(sessionToken)) {
-            getLocalSessionToken(request, sessionToken, StringUtils.EMPTY);
+            String partitionKeyRangeId = request.requestContext.resolvedPartitionKeyRange.getId();
```

Comment:
Wow, that was a bug?

Reply:
this is for validation. Why this change is needed?

Reply:
Yes, this was a bug. 
It is guaranteed, if not present, then it is a bug.

Reply:
This code path will always throw IllegalStateException - because we are always passing empty string - here is the issue - https://github.com/Azure/azure-sdk-for-java/issues/9053

Reply:
@kushagraThapar https://github.com/Azure/azure-sdk-for-java/issues/9053 is for the PartitionKey.None is that right link?

Reply:
Yes, my bad. Here is the right link - https://github.com/Azure/azure-sdk-for-java/issues/10493

Reply:
@kirankumarkolli .Net seem to be doing the same thing as we used to do here:

SessionTokenHelper.cs:
```.Net
public static void ValidateAndRemoveSessionToken(DocumentServiceRequest request)
{
    string sessionToken = request.Headers[HttpConstants.HttpHeaders.SessionToken];
    if (!string.IsNullOrEmpty(sessionToken))
    {
        GetLocalSessionToken(request, sessionToken, string.Empty);
         request.Headers.Remove(HttpConstants.HttpHeaders.SessionToken);
    }
}
```

Reply:
True, but .Net doesn't throw Exception on passing empty partition key range id. Like we do: 

```
if (partitionKeyRangeId == null || partitionKeyRangeId.isEmpty()) {
            // AddressCache/address resolution didn't produce partition key range id.
            // In this case it is a bug.
            throw new IllegalStateException("Partition key range Id is absent in the context.");
        }
```

May be we should get rid of this exception clause then, instead of passing partitionId and keep passing empty string to be in parity with .Net ?

Reply:
Discussed offline with @moderakh 
Final logic is, we will just get rid of `partitionKeyRangeId.isEmpty()` check.